src/QsCompiler/Tests.Compiler/TestCases/TypeChecking.qs

```diff
@@ -737,4 +737,85 @@ namespace Microsoft.Quantum.Testing.TypeChecking {
     function SizedArrayInvalid3() : Int[] {
         return [5, size = (1, 2)];
     }
+
+    // Lambdas
+
+    function Lambda1<'a>() : 'a -> 'a {
+        return x -> x;
+    }
```

Comment:
Just Curiosity: What is `x` in this scope? To me it is an undefined identifier.

Reply:
The syntax for lambda expressions is `pattern ('->' | '=>') expression`, so the first `x` is a pattern binding the entire lambda parameter to the symbol `x`. The second `x` references this binding.